### PR TITLE
Add pytorch_jit sample Colab notebook using SHARK-Turbine. 

### PR DESCRIPTION
**Pull request by ScottTodd**
Progress on https://github.com/openxla/iree/issues/15117

This notebook shows how to use [SHARK-Turbine](https://github.com/nod-ai/SHARK-Turbine) for eager execution within a PyTorch session using IREE and [torch-mlir](https://github.com/llvm/torch-mlir) under the covers. I'm starting simple to get the concepts across, with minimal API usage and a tiny `nn.Module` sourced from https://pytorch.org/docs/stable/notes/modules.html. My expectation is that this notebook will evolve alongside other notebooks (e.g. `pytorch_aot.ipynb`), documentation (https://github.com/openxla/iree/issues/15114), and the SHARK-Turbine project itself.

Preview URL for review: https://colab.research.google.com/github/openxla/iree/blob/scotttodd-pytorch-samples-1/samples/colab/pytorch_jit.ipynb

skip-ci: no-op

### CODE REVIEW

File: samples/colab/pytorch_jit.ipynb
Comment (ScottTodd) [inline samples/colab/pytorch_jit.ipynb]: https://github.com/openxla/iree/actions/runs/6462150219/job/17543263981#step:3:309

Looks like Turbine requires Python > 3.10 (https://docs.python.org/3/library/types.html#types.NoneType)
```python
File /work/.notebook.venv/lib/python3.9/site-packages/shark_turbine/dynamo/importer.py:10
      8 import operator
      9 import re
---> 10 from types import NoneType, BuiltinMethodType, BuiltinFunctionType
     11 from typing import Any, Callable, Dict, List, Optional, Sequence, Set, Tuple, Union
     12 import numpy as np

ImportError: cannot import name 'NoneType' from 'types' (/usr/lib/python3.9/types.py)
```

Our Docker image installs 3.9: https://github.com/openxla/iree/blob/77a8741a9eb5a03404bcbb3343f90ff250d11099/build_tools/docker/dockerfiles/base.Dockerfile#L61-L62

I should just update us to use Colab's Dockerfile instead: https://github.com/openxla/iree/issues/15070

File: samples/colab/README.md
Comment (ScottTodd) [inline samples/colab/README.md:13]: May rename these notebooks to `tensorflow_` or `tf_` to make it easier to see the alphabetical ordering and grouping.

File: samples/colab/pytorch_jit.ipynb
Comment (ScottTodd) [reply samples/colab/pytorch_jit.ipynb]: Docker updates are proving to be tricky... skipping test coverage for now.

File: samples/colab/README.md
Comment (ScottTodd) [reply samples/colab/README.md:13]: Will leave that to a follow-up, since more files will need to be updated as part of renaming.

Comment (stellaraccident) [reply samples/colab/README.md:13]: Or remove them. In practice we don't have a maintainer for tensorflow and it hurts every time we have to bake an excuse to people.

**Review by stellaraccident — APPROVED** [review]
Thanks. I don't really know how to review colab so will look more once landed and further along.

**Comment by ScottTodd** [comment]
> Thanks. I don't really know how to review colab so will look more once landed and further along.

I left a link to view in Colab in the PR description: https://colab.research.google.com/github/openxla/iree/blob/scotttodd-pytorch-samples-1/samples/colab/pytorch_jit.ipynb

For commenting, you can toggle between the source and "rich diff" views in GitHub: 
image

File: samples/colab/README.md
Comment (ScottTodd) [reply samples/colab/README.md:13]: Rename is in https://github.com/openxla/iree/pull/15155

Yeah... I feel the maintenance and support pain... maybe I'll remove or simplify the existing notebooks after standing up new ones in PyTorch (and ideally also JAX).

**Pull request merged**
